code/model/modelanimation.cpp

```diff
@@ -6,23 +6,23 @@ extern float flFrametime;
 
 namespace animation {
 
-	std::multimap<ship*, std::shared_ptr<ModelAnimation>> ModelAnimation::s_runningAnimations;
+	std::multimap<polymodel_instance*, std::shared_ptr<ModelAnimation>> ModelAnimation::s_runningAnimations;
```

Comment:
I think these maps are going to persist for more than one frame.  In this case, the map indexes should be the integer id (shipnum or objnum, and pmi->id).  The `Polymodel_instance` collection is a vector, and the `Ships` collection may (someday) also be a vector.  Vectors may re-allocate their storage when their size increases, which means that all pointers will be invalidated.